packages/operator/pkg/apiserver/routes/v1/packaging/model_packaging_validation.go

```diff
@@ -107,10 +107,13 @@ func (mpv *MpValidator) validateMainParameters(mp *packaging.ModelPackaging) (er
 
 	err = multierr.Append(err, validation.ValidateID(mp.ID))
 
+	// If image is empty then set image from Packaging Integration
 	if len(mp.Spec.Image) == 0 {
-		packagingIntegration, k8sErr := mpv.piRepo.GetPackagingIntegration(mp.Spec.IntegrationName)
-		if k8sErr != nil {
-			err = multierr.Append(err, k8sErr)
+		packagingIntegration, getPiErr := mpv.piRepo.GetPackagingIntegration(mp.Spec.IntegrationName)
+		if getPiErr != nil {
+			logMP.Info(
+				"Unable to extract default image from Packaging integration because it is empty",
+			)
```

Comment:
```suggestion
		if len(mp.Spec.IntegrationName) == 0 {
			err = multierr.Append(err, errors.New("Model Packaging must have either Image or PackagingIntegration"))
		}
		packagingIntegration, getPiErr := mpv.piRepo.GetPackagingIntegration(mp.Spec.IntegrationName)
		if getPiErr != nil {
			logMP.Error(getPiErr, "Unable to get Packaging integration from repository")
			panic(getPiErr)
```
1. I suggest to explicitly check if `mp.Spec.IntegrationName` is empty, because it is not the only reason the repository can return error. 
2. When we know that `mp.Spec.IntegrationName` is **not empty** and still we get error from repo, it is really bad unexpected condition of the system. Basically it is 500 http status, so we can either `panic`, or handle this case in other way you like. 
What do you think?

Reply:
1. I agree that we should check error using .IsNotFoundError()